### PR TITLE
Multi LLM prompt studio fixes

### PR DESCRIPTION
**Pull request by jagadeeswaran-zipstack**
## What

- This PR focuses on stabilizing Prompt studio Multi LLM.

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- No, it has changes on Multi LLM.

## Database Migrations

- N/A

## Env Config

- N/A

## Relevant Docs

- N/A

## Related Issues or PRs

- N/A

## Dependencies Versions

- N/A

## Notes on Testing

-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Review by chandrasekharan-zipstack — APPROVED** [review]
BE changes LGTM - left some nit comments on this code in #513 which could be discussed and incorporated

File: frontend/src/components/custom-tools/output-for-doc-modal/OutputForDocModal.jsx [hunk 1]
```diff
@@ -117,13 +118,63 @@ function OutputForDocModal({
 
   const updatePromptOutput = (data) => {
     setPromptOutputs((prev) => {
-      // If data is provided, use it; otherwise, create a copy of the previous state
-      const updatedPromptOutput = data || [...prev];
+      const updatedPromptOutput = getUpdatedPromptOutput(data, prev);
```
Comment (vishnuszipstack) [inline frontend/src/components/custom-tools/output-for-doc-modal/OutputForDocModal.jsx:121]: @jagadeeswaran-zipstack why this changed to function call?

**Review by vishnuszipstack — APPROVED** [review]

File: frontend/src/components/custom-tools/output-for-doc-modal/OutputForDocModal.jsx (hunk 1 above)
Comment (jagadeeswaran-zipstack) [reply frontend/src/components/custom-tools/output-for-doc-modal/OutputForDocModal.jsx:121]: Refactored code to reduce nesting.

File: backend/prompt_studio/prompt_studio_output_manager/output_manager_helper.py [hunk 2]
```diff
@@ -144,3 +145,39 @@ def fetch_default_llm_profile(tool: CustomTool) -> ProfileManager:
...
+                    is_single_pass_extract=False,
+                    document_manager_id=document_manager_id,
+                )
+
+                if not queryset.exists():
+                    result[tool_prompt.prompt_key] = ""
+                    continue
+
+                for output in queryset:
+                    result[tool_prompt.prompt_key] = output.output
```
Comment (harini-venkataraman) [inline backend/prompt_studio/prompt_studio_output_manager/output_manager_helper.py:180]: @jagadeeswaran-zipstack Can you add your inputs to the [discussion](https://github.com/Zipstack/unstract/pull/513#discussion_r1689070317)?

**Review by Deepak-Kesavan — APPROVED** [review]

**Pull request merged**